### PR TITLE
adding support for multiple gpgkeys & removing neutron-sriov-nic-agent package

### PR DESCRIPTION
**Pull request by psairam**
This change has the support for following:
PROD-10309: Multiple Signed packages for CX5 deployment.
OPENSTACK-2576: Removing neutron-sriov-nic-agent package name from NUAGE_DEPENDENCIES.
PROD-10136: No longer patching puppet files on overcloud puppet codebase.

### CODE REVIEW

File: image-patching/stopgap-script/nuage_overcloud_full_patch.py [hunk 1]
```diff
@@ -284,95 +204,105 @@ def update_kernel(image):
...
+            logger.error(
+                "Nuage package signing key is not present in %s ,Installation cannot proceed.  Please place the signing key in the correct location and retry" % (
+                i[0]))
+            sys.exit(1)
+
+
+
+def image_patching(args):
+
+    if args.logFile:
```
Comment (sunny-verma) [inline image-patching/stopgap-script/nuage_overcloud_full_patch.py]: make it a default behavior

File: image-patching/stopgap-script/nuage_overcloud_full_patch.py [hunk 2]
```diff
@@ -284,95 +204,105 @@ def update_kernel(image):
...
+        handler = logging.FileHandler(args.logFile)
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
 
-        cmds_run(['echo "Updating Kernel"'])
-        update_kernel(argsDict['ImageName'][0])
+    GPGCHECK = 1
```
Comment (sunny-verma) [inline image-patching/stopgap-script/nuage_overcloud_full_patch.py]: Please move all static variable at top of the file.

File: image-patching/stopgap-script/nuage_overcloud_full_patch.py [hunk 3]
```diff
@@ -284,95 +204,105 @@ def update_kernel(image):
...
-                                  argsDict['ImageName'][0])
-
-        cmds_run(['echo "Uninstalling packages"'])
-        uninstall_packages(argsDict['ImageName'][0], argsDict['Version'][0])
-
-        cmds_run(['echo "Creating Repo File"'])
-        if 'RepoName' in argsDict:
-            create_repo_file(argsDict['RepoName'][0], argsDict['RepoBaseUrl'][0], argsDict['ImageName'][0])
+def importing_gpgkeys(image, workingDir, gpgkeys):
+    for i in gpgkeys:
```
Comment (sunny-verma) [inline image-patching/stopgap-script/nuage_overcloud_full_patch.py]: change i to gpgkey

File: image-patching/stopgap-script/nuage_overcloud_full_patch.py [hunk 4]
```diff
@@ -165,30 +201,41 @@ def create_repo_file(reponame, repoUrl, image, gpgcheck):
...
 
 #####
 # Function to clean up the repo file
 #####
 def delete_repo_file(image):
-    virt_customize('"rm -f /etc/yum.repos.d/nuage.repo" -a %s --selinux-relabel' % (image))
+    virt_customize(
+        '"rm -f /etc/yum.repos.d/nuage.repo" -a %s --selinux-relabel'
+        % (
+            image))
```
Comment (sunny-verma) [inline image-patching/stopgap-script/nuage_overcloud_full_patch.py]: why image is on seperate line ? it can be changed to `% (image))`

Comment (psairam) [reply image-patching/stopgap-script/nuage_overcloud_full_patch.py]: done

**Pull request merged**